### PR TITLE
Preliminary adaptive resolution

### PR DESCRIPTION
**Pull request by bprather**
* All the nice re-organization preliminaries
* Probably doesn't leak too much memory
* Heuristic for refinement may need to be refined

### CODE REVIEW

**Comment by bprather** [comment]
This is currently blocked on:
1. Slow light test, to make sure I haven't broken it
2. Adaptive res regression test, to ensure adaptive images never look truly crazy compared to full-res
3. Knowing what to do w.r.t. interpolation.  Currently uses straight bi-linear, but interpolating from interpolated pixels is fuzzy.  May need to keep track of calculated vs interpolated pixels & use just the former.
The heuristic could always use work, suggestions welcome...

**Comment by bprather** [comment]
This branch has progressed far enough to be merged I think.  Quick summary:
1. New heuristic from Zack Gelles
2. Regression test that a suitably stringent refinement requirement produces an image with very low MSE.  Note that compare.py return criterion is now MSE, it was relative flux difference for some reason.
3. While there's still work in figuring out interpolation, the code already tracks which pixels were traced vs interpolated, so it is unlikely to need major changes
In short this solves blocker (2) and lays a good foundation for (3).  (1) will take longer and has its own bug, #24
